pkg/chunk/purger/blocks_purger_api.go

```diff
@@ -13,28 +14,38 @@ import (
 	"github.com/prometheus/client_golang/prometheus"
 	"github.com/thanos-io/thanos/pkg/objstore"
 
+	"github.com/cortexproject/cortex/pkg/ruler/rules"
 	"github.com/cortexproject/cortex/pkg/storage/bucket"
 	cortex_tsdb "github.com/cortexproject/cortex/pkg/storage/tsdb"
 	"github.com/cortexproject/cortex/pkg/tenant"
 	"github.com/cortexproject/cortex/pkg/util"
 )
 
+type TenantDeletionConfig struct {
+	EnableDeletionOfRuleGroups bool `yaml:"rule_groups_enabled"`
```

Comment:
A couple of feedback on the config:
1. What's the use case to enable it conditionally? Do we really need this config?
1. I would make this config a child of the purger config and not a root config entry

Reply:
> 1. What's the use case to enable it conditionally? Do we really need this config?

Reason for introducing this option is that not all rule-stores allow deletion of rule groups, and instead return error for any attempt. To make tenant deletion API usable when using such rule stores, it is possible to disable deleting of rule groups.

> 2. I would make this config a child of the purger config and not a root config entry
Done.

Reply:
> Reason for introducing this option is that not all rule-stores allow deletion of rule groups, and instead return error for any attempt. To make tenant deletion API usable when using such rule stores, it is possible to disable deleting of rule groups.

Can we auto-detect it?

Reply:
Yes, we can. I've changed the code to do that now, and removed the option. PTAL.